### DISCUSSION TITLE
Add support for expressions

### DISCUSSION THREAD

**Discussion by rickstaa** [grafana/grafana-infinity-datasource#286] [Q&A] [answered]
For my Grafana dashboard, I am trying to fetch some data from [this API endpoint](https://api-eu.livepool.io/api/v1/status). The response returned by this endpoint is as follows:

```JSON
{
  "Commission": "5.00",
  "Version": "0.5.28-4a06ed71",
  "BasePrice": "1150",
  "TotalPayouts": "1237257813626346396"
}
```

I am able to fetch the `TotalPayouts` as follows:

![image](https://user-images.githubusercontent.com/17570430/159439982-ac869661-b283-4d59-91de-0e1b17c5aec3.png)

The problem with the `TotalPayouts` variable is that it is displayed in Wei, where I want to display ETH. The conversion between these two variables is quite straightforward `Wei*1e-18`. I, however, can not find a way to do this with the current Infinity plugin.

I tried using the `Add Field From Calculation` transform, but this only allows `Reduce Row` (i.e. `max`, `min`, `last`, `first`, `mean` etc.) and `Binary` operations with existing fields. 

I, therefore, think that giving users the ability to do expressions like this would be a significant improvement.

![image](https://user-images.githubusercontent.com/17570430/159440860-807cb490-d278-4992-99e0-1709d895c45a.png)

**Comment by yesoreyeram** [comment, marked as answer]
Have you tried UQL? You can do mathematical operations there.

**Reply by rickstaa** [reply]
Great,

```uql 
project BasePrice
```

does the trick thanks!
